### PR TITLE
Fix incorrect line/line intersection on `aarch64`

### PR DESCRIPTION
**Pull request by peterstace**
## Description

Fixes a bug where the result of intersecting two lines segments is sometimes incorrect.

The bug only occurs in `aarch64` and is due to an unanticipated [fused multiply-add](https://en.wikipedia.org/wiki/Multiply%E2%80%93accumulate_operation) (FMA) operation. The FMA causes the cross product method to return very small but non-zero results for linearly dependent inputs (i.e. two vectors that are scaled versions of each other).

The bug does **not** occur on `x64_64`.

## Check List

Have you:

- Added unit tests? Yes.

- Add cmprefimpl tests? (if appropriate?) N/A.

- Updated release notes? (if appropriate?)

## Related Issue

- https://github.com/peterstace/simplefeatures/issues/364

### CODE REVIEW

File: geom/xy_test.go [hunk 1]
```diff
@@ -52,3 +53,40 @@ func TestXYUnit(t *testing.T) {
...
+	}
+	for i, tc := range []struct {
+		u, v geom.XY
+		want float64
+	}{
+		// Contains zero:
+		{u: xy(1, 1), v: xy(0, 0), want: 0},
+		{u: xy(0.1, 0.1), v: xy(0, 0), want: 0},
+
+		// Not linearly independent:
```
Comment (albertteoh) [inline geom/xy_test.go:69]: Maybe could reword this as `Linearly dependent:`?

**Review by albertteoh — APPROVED** [review]
Nice discovery!

File: geom/xy_test.go (hunk 1 above)
Comment (peterstace) [reply geom/xy_test.go]: Good suggestion, altered in 51c7c7327f57a19bf873a4133046c5c16e22e8d5.

**Comment by peterstace** [comment]
Thanks for reviewing!
